test/NumSharp.UnitTest/README.md

```diff
@@ -7,4 +7,52 @@ Install `Codecov` from https://www.nuget.org/packages/Codecov.
 Add `.nuget\packages\codecov\1.1.0\tools` to system bin Path
 
 `PS D:\Projects\NumSharp> OpenCover.Console -register:user -target:"dotnet.exe" -targetargs:test -filter:"+[NumSharp.UnitTest]*" -output:"coverage.xml" -oldstyle`
-`PS D:\Projects\NumSharp> codecov -f coverage.xml -t <codecov-numsharp-token>`
\ No newline at end of file
+`PS D:\Projects\NumSharp> codecov -f coverage.xml -t <codecov-numsharp-token>`
+
+
+### NDArray generation for Unit Test cases
+
+Generating NDArray constants by hand is tedious and error prone, so here
+is some Python code that spits out C# definitions for static NDArrays
+that can be used in unit tests:
+
+```python
+import math
+import numpy as np
+
+class cSharp:
+    def asCode2D(varName, v):
+        if v.dtype.name == "int32":
+            vType = "Int32"
+        elif v.dtype.name == "float64":
+            vType = "double"
+        else:
+            vType = "unknown"
+        print("        static NDArray {0} = new NDArray(new {1}[] {{".format(varName, vType))
+        valstr = ""
+        commasToPrint = v.shape[0] * v.shape[1] - 1
+        for i, row in enumerate(v):
+            rowStr = "            "
+            for j, item in enumerate(row):
+                rowStr = rowStr + "{}".format(item)
+                if commasToPrint > 0:
+                    rowStr = rowStr + ", "
+                commasToPrint -= 1
+                #if (i < v)
+            print(rowStr)
+        print("            }}, new Shape(new int[] {{ {}, {} }}));".format(v.shape[0], v.shape[1]))
+        print("")
+
+# generate mgrid unit test data (we're trusting Python here)
+aa, bb = np.mgrid[0:5, 0:3]
```

Comment:
The code that was used to generate you can place as a comment in `Creation/NdArray.Mgrid.Test.cs`